### PR TITLE
[MAINT][DOC] add action to test on matlab

### LINKED ISSUE #85: Improve doc to add CI setup examples to run with github
I have a template repo and a cookie cutter with things I could extract to add to the doc here describing how to do it. 

https://github.com/Remi-Gau/template_matlab_analysis

Let me know it there is some interest.

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #85 

- adds a github workflow to run the tests on github:
  - with matlab on all OS with several matlab versions
  - with octave
- update readme on how to run tests on github CI
  - with octave using the moxunit github action
  - with matlab
- add dependabot config to keep actions up to date

## TODO 
- [x] add mocov to example

### CODE REVIEW

**Comment by Remi-Gau** [comment]
@nno this should be ready for a first review.

This CI section in the README is getting quite long, so I have second thoughts about adding another way to test things with Octave on Github.

Let me know what you think.

**Comment by nno** [comment]
> This CI section in the README is getting quite long, so I have second thoughts about adding another way to test things with Octave on Github.

Actually I think this would be useful, so if you could add something I would appreciate that.

**Comment by nno** [comment]
@Remi-Gau I noticed this PR is still open. In its present state there is a merging conflict. What would need to be done to make this branch ready for a merge into master?

**Comment by Remi-Gau** [comment]
will fix conflict and do some more cleaning

**Comment by Remi-Gau** [comment]
wonder if it makes sense to have the workflow with the moxunit github action?

not sure what it achieves that the matlab and octave workflow don't already do

if we want to test the moxunit github action it should probably go in the repo of the action

**Comment by Remi-Gau** [comment]
@jorgepz @nno 

FYI this PR sets up coverage with codecov but I can set up coveralls it that's better.

Questions: 
- should this replace the travis tests?
- I see that the travis test also install modox: should we test modox here or in its dedicated repo?

**Comment by jorgepz** [comment]
I am used to codecov but anything free that works is fine by me.
IMHO it is better to test modox in its repo

File: .github/workflows/CI_matlab..yml
Comment (jorgepz) [inline .github/workflows/CI_matlab..yml:1]: thanks for the ymls @Remi-Gau ! I think this file should be renamed to remove an additional "."

**Review by jorgepz — APPROVED** [review]

**Pull request merged**